app/grandchallenge/reader_studies/tasks.py

```diff
@@ -20,38 +18,38 @@
 )
 
 
-@transaction.atomic
-def add_score(obj, answer):
-    obj.calculate_score(answer)
-    obj.save(update_fields=["score"])
-
-
-@transaction.atomic
 def add_image(obj, image):
     obj.answer_image = image
     obj.save()
     image.assign_view_perm_to_creator()
     image.update_viewer_groups_permissions()
 
 
-@acks_late_micro_short_task
-def add_scores_for_display_set(*, instance_pk, ds_pk):
-    instance = Answer.objects.get(pk=instance_pk)
-    display_set = DisplaySet.objects.get(pk=ds_pk)
-    if instance.is_ground_truth:
-        for answer in Answer.objects.filter(
-            question=instance.question,
-            is_ground_truth=False,
-            display_set=display_set,
-        ):
-            add_score(answer, instance.answer)
-    else:
-        ground_truth = Answer.objects.filter(
-            question=instance.question,
-            is_ground_truth=True,
-            display_set=display_set,
-        ).get()
-        add_score(instance, ground_truth.answer)
+@acks_late_2xlarge_task
+@transaction.atomic
+def answers_from_ground_truth(*, reader_study_pk, target_user_pk):
+    reader_study = ReaderStudy.objects.get(pk=reader_study_pk)
+    target_user = get_user_model().objects.get(pk=target_user_pk)
+
+    all_answers = Answer.objects.filter(question__reader_study=reader_study)
+
+    for answer in all_answers.filter(is_ground_truth=True):
+        # Simplify permissions and create new answers
+        answer._state.adding = True
+        answer.id = None
+        answer.pk = None
+        answer.is_ground_truth = False
+        answer.creator = target_user
+        Answer.validate(
+            creator=answer.creator,
+            question=answer.question,
+            answer=answer.answer,
+            display_set=answer.display_set,
+            is_ground_truth=answer.is_ground_truth,
+        )
+        answer.save(calculate_score=False)
+
+    all_answers.update(score=None)
```

Comment:
Why are the scores reset here? The ground truth still exists so the answers should still have scores, right? I would have thought that they would only need to be reset when the ground truth is removed.

Reply:
Ah, that is a left over from the initial logic where we were actively destroying ground truth instead of copying it. Removed it now and also removed the associated tests snippets.